CMakeLists.txt

```diff
@@ -1,4 +1,12 @@
-cmake_minimum_required(VERSION 3.2)
+cmake_minimum_required(VERSION 3.20)
+project(filepattern)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+set (filepattern_VERSION_MAJOR 2)
+set (filepattern_VERSION_MINOR 0)
+set (filepattern_VERSION_BUILD 0)
+set(filepattern_VERSION "${filepattern_VERSION_MAJOR}.${filepattern_VERSION_MINOR}.${filepattern_VERSION_BUILD}")
```

Comment:
Is there any way to get the version number from versioneer/github or does that not work in CMake?

Reply:
I added a version.h file similar to Nyxus. IMO, trying to use Git to extract version info adds an extra dependency to have Git installed on the system to build from source. In theory, a zipped source can be used to build it w/o git.